go/libraries/doltcore/schema/typeinfo/typeinfo.go

```diff
@@ -252,18 +275,47 @@ func FromSqlType(sqlType sql.Type) (TypeInfo, error) {
 		if !ok {
 			return nil, fmt.Errorf(`expected "EnumTypeIdentifier" from SQL basetype "Enum"`)
 		}
-		return &enumType{enumSQLType}, nil
+		enumSqlType, err := fillInEnumCollationWithDefault(enumSQLType)
+		if err != nil {
+			return nil, err
+		}
+		return &enumType{enumSqlType}, nil
 	case sqltypes.Set:
 		setSQLType, ok := sqlType.(sql.SetType)
 		if !ok {
 			return nil, fmt.Errorf(`expected "SetTypeIdentifier" from SQL basetype "Set"`)
 		}
+		setSQLType, err := fillInSetCollationWithDefault(setSQLType)
+		if err != nil {
+			return nil, err
+		}
 		return &setType{setSQLType}, nil
 	default:
 		return nil, fmt.Errorf(`no type info can be created from SQL base type "%v"`, sqlType.String())
 	}
 }
 
+func fillInSetCollationWithDefault(st sql.SetType) (sql.SetType, error) {
+	if st.Collation() == sql.Collation_Unspecified {
```

Comment:
Far better way to do this:
```go
func someFuncName(typ sql.Type) (sql.Type, error) {
	if collationType, ok := typ.(sql.TypeWithCollation); ok && collationType.Collation() == sql.Collation_Unspecified {
		return collationType.WithNewCollation(sql.Collation_Default)
	}
	return typ, nil
}
```

Reply:
Instead of making this a function, this logic could also go at the top of `FromSqlType` and it'll automatically apply to all types. That `queryType := sqlType.Type()` line could do this instead. Then, any added types that handle collations won't have a coverage gap.

Reply:
Good idea